projects/core/src/modules/scrollable-host/scrollable-host.service.ts

```diff
@@ -14,30 +14,109 @@ export class SkyScrollableHostService {
     private windowRef: SkyAppWindowRef
   ) { }
 
-  public getScrollabeHost(elementRef: ElementRef): HTMLElement | Window {
+  /**
+   * Returns the given element's current scrollable host
+   * @param elementRef The element whose scrollable host is being requested
+   * @returns The current scrollable host
+   */
+  public getScrollableHost(elementRef: ElementRef): HTMLElement | Window {
     return this.findScrollableHost(elementRef.nativeElement);
   }
 
-  public watchScrollableHost(elementRef: ElementRef, completionObservable: Observable<void>): Observable<HTMLElement | Window> {
-    let scrollableHost = this.findScrollableHost(elementRef.nativeElement);
-    let behaviorSubject = new BehaviorSubject(scrollableHost);
-
-    const mutationObserver = this.mutationObserverSvc.create(() => {
-      let newScrollableHost = this.findScrollableHost(elementRef.nativeElement);
-
-      if (newScrollableHost !== scrollableHost) {
-        scrollableHost = newScrollableHost;
+  /**
+   * Returns an observable which emits the given element's current scrollable host
+   * @param elementRef The element whose scrollable host is being requested
+   * @param completionObservable An observable which alerts the internal observers that they should complete
+   * @returns An observable which emits the current scrollable host
+   * @internal
+   */
+  public watchScrollableHost(elementRef: ElementRef): Observable<HTMLElement | Window> {
+    let subCount = 0;
```

Comment:
Since you're caching the list of subscribers in an array and presumably removing them from the array when unsubscribing (see my feedback below on that), I'd suggest checking the array's length rather than keeping a separate counter.